tests/test_snapshot_download.py

```diff
@@ -88,3 +106,29 @@ def test_download_model(self):
 
             # folder name contains the revision's commit sha.
             self.assertTrue(self.first_commit_hash in storage_folder)
+
+    def test_download_private_model(self):
+        # Test download with token
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            storage_folder = snapshot_download(
+                f"{USER}/{REPO_NAME_PRIVATE}",
+                revision="main",
+                use_auth_token=True,
+                cache_dir=tmpdirname,
+            )
+            folder_contents = os.listdir(storage_folder)
+            self.assertEqual(len(folder_contents), 2)
+            self.assertTrue("dummy_file.txt" in folder_contents)
+            self.assertTrue(".gitattributes" in folder_contents)
+
+            with open(os.path.join(storage_folder, "dummy_file.txt"), "r") as f:
+                contents = f.read()
+                self.assertEqual(contents, "v1")
+
+            # folder name contains the revision's commit sha.
+            self.assertTrue(self.private_commit_hash in storage_folder)
+
+        # Test download fails without token
+        self._api.logout(self._token)
```

Comment:
This feels a bit hacky, but is needed because we define a `token` for all tests in `SnapshotDownloadTests`. The alternative would be to pass a dummy token like `api_XXX` to `snapshot_download`, but that doesn't test the default scenario where `use_auth_token=None`. Finally we could move these private repo tests to another test class, but that seems like overkill IMO